src/main/services/transactions/transaction_service.ts

```diff
@@ -0,0 +1,96 @@
+import { Resource } from "@companieshouse/api-sdk-node";
+import { Transaction } from "@companieshouse/api-sdk-node/dist/services/transaction/types";
+import logger from "../../../../lib/Logger";
+import { createPublicOAuthApiClient } from "../api/api_service";
+import { Session } from "@companieshouse/node-session-handler";
+import ApiClient from "@companieshouse/api-sdk-node/dist/client";
+import { ApiErrorResponse, ApiResponse } from "@companieshouse/api-sdk-node/dist/services/resource";
+import { StatusCodes } from "http-status-codes";
+
+import { CREATE_DESCRIPTION, REFERENCE, transactionStatuses } from "../../config";
+
+/**
+ * Post transaction
+ */
+export const postTransaction = async (session: Session, companyNumber: string, description: string, reference: string): Promise<Transaction> => {
+    const apiClient: ApiClient = createPublicOAuthApiClient(session);
+
+    const transaction: Transaction = {
+        companyNumber,
+        reference,
+        description
+    };
+
+    logger.debug(`Creating transaction with company number ${companyNumber}`);
+    const sdkResponse: Resource<Transaction> | ApiErrorResponse = await apiClient.transaction.postTransaction(transaction);
+
+    if (!sdkResponse) {
+        logger.error(`Transaction API POST request returned no response for company number ${companyNumber}`);
+        return Promise.reject(sdkResponse);
+    }
+
+    if (!sdkResponse.httpStatusCode || sdkResponse.httpStatusCode !== StatusCodes.CREATED) {
+        logger.error(`Http status code ${sdkResponse.httpStatusCode} - Failed to post transaction for company number ${companyNumber}`);
+        return Promise.reject(sdkResponse);
+    }
+
+    const castedSdkResponse: Resource<Transaction> = sdkResponse as Resource<Transaction>;
+
+    if (!castedSdkResponse.resource) {
+        logger.error(`Transaction API POST request returned no resource for company number ${companyNumber}`);
+        return Promise.reject(sdkResponse);
+    }
+
+    logger.debug(`Received transaction ${JSON.stringify(sdkResponse)}`);
+
+    return Promise.resolve(castedSdkResponse.resource);
+};
+
+/**
+ * Close transaction
+ */
+export const closeTransaction = async (session: Session, companyNumber: string, transactionId: string, objectId: string|undefined): Promise<ApiResponse<Transaction>> => {
+    const apiResponse: ApiResponse<Transaction> = await putTransaction(session, companyNumber, transactionId, CREATE_DESCRIPTION, transactionStatuses.CLOSED, objectId).catch((sdkResponse) => {
+        return Promise.reject(sdkResponse);
+    });
+    return Promise.resolve(apiResponse);
+};
+
+/**
+ * PUT transaction
+ */
+export const putTransaction = async (session: Session,
+    companyNumber: string,
```

Comment:
put transaction is also been developed in acsp-api

Reply:
Will work on this when working on Transaction model